src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/perform_interaction_test.cc

```diff
@@ -70,35 +70,109 @@ namespace strings = ::application_manager::strings;
+    choice_set2[strings::choice_set][0][strings::choice_id] = kVrChoiceID;
+
+    choice_set_map_[kCorrelationId].insert(std::make_pair(
+        kUiChoiceID, new smart_objects::SmartObject(choice_set1)));
+    choice_set_map_[kCorrelationId].insert(std::make_pair(
+        kVrChoiceID, new smart_objects::SmartObject(choice_set2)));
+
     ON_CALL(app_mngr_, application(kConnectionKey))
         .WillByDefault(Return(mock_app_));
     ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+    ON_CALL(*mock_app_, performinteraction_choice_set_map())
+        .WillByDefault(Return(DataAccessor<app_mngr::PerformChoiceSetMap>(
+            choice_set_map_, performinteraction_choice_set_lock_ptr_)));
   }
 
   void ResultCommandExpectations(MessageSharedPtr msg,
+                                 bool success,
+                                 hmi_apis::Common_Result::eType result_code,
                                  const std::string& info) {
     EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::success].asBool(),
-              true);
-    EXPECT_EQ(
-        (*msg)[am::strings::msg_params][am::strings::result_code].asInt(),
-        static_cast<int32_t>(hmi_apis::Common_Result::UNSUPPORTED_RESOURCE));
+              success);
+    EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::result_code].asInt(),
+              static_cast<int32_t>(result_code));
     EXPECT_EQ((*msg)[am::strings::msg_params][am::strings::info].asString(),
               info);
   }
 
+  void HMIRequestExpectations(MessageSharedPtr msg,
+                              hmi_apis::FunctionID::eType function_id,
+                              const std::string& method_name) {
+    EXPECT_EQ((*msg)[am::strings::params][am::strings::function_id].asInt(),
+              static_cast<int32_t>(function_id));
+    EXPECT_EQ((*msg)[am::strings::msg_params][am::hmi_request::method_name]
+                  .asString(),
+              method_name);
+  }
+
+  MessageSharedPtr CreateRequestMessage(
+      const mobile_apis::InteractionMode::eType& interaction_mode) {
+    MessageSharedPtr request_msg = CreateMessage(smart_objects::SmartType_Map);
+    (*request_msg)[strings::params][strings::connection_key] = kConnectionKey;
+    (*request_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*request_msg)[strings::msg_params][strings::interaction_mode] =
+        interaction_mode;
+    return request_msg;
+  }
+
+  MessageSharedPtr CreateHMIResponseMessage(
+      const hmi_apis::Common_Result::eType& response_code,
+      const std::string& message_info,
+      const int32_t command_id = kCommandId) {
+    MessageSharedPtr response_msg = CreateMessage(smart_objects::SmartType_Map);
+    (*response_msg)[strings::params][hmi_response::code] = response_code;
+    (*response_msg)[strings::msg_params][strings::cmd_id] = command_id;
+    (*response_msg)[am::strings::msg_params][am::strings::info] = message_info;
```

Comment:
Why mix `am::strings` and `strings::`? I know old code uses `am::` but please use `strings::` in your new code.

Reply:
@iCollin fixed in https://github.com/smartdevicelink/sdl_core/pull/3122/commits/8e788552cae05a5fad4291e0fd234291f7d2e390